edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/MetricFilter.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.Diagnostics
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using Newtonsoft.Json;
+
+    /// <summary>
+    /// This class acts as a pass through for a group of metrics.
+    /// It will exclude metrics based on a tag whitelist and blacklist.
```

Comment:
I get the idea, but having both a whitelist and blacklist seems odd somehow. Can't we do just with a whitelist?

Reply:
I'm only using the whitelist, but I wanted to keep it generic. Hopefully this will be re-used when we expose it to customers, so giving them a robust filter is good.

Reply:
Removed blacklist to make it simple. It is easy to add in later if needed